core/src/main/scala/org/apache/spark/resource/ResourceAllocator.scala

```diff
@@ -38,11 +37,13 @@ private[spark] trait ResourceAllocator {
    * For task resources ([[org.apache.spark.scheduler.ExecutorResourceInfo]]), this value
    * can be a multiple, such that each address can be allocated up to [[slotsPerAddress]]
    * times.
-   *
-   * TODO Use [[org.apache.spark.util.collection.OpenHashMap]] instead to gain better performance.
    */
   private lazy val addressAvailabilityMap = {
-    mutable.HashMap(resourceAddresses.map(_ -> slotsPerAddress): _*)
+    val map: OpenHashMap[String, Int] = new OpenHashMap[String, Int]()
```

Comment:
This is a `lazy val`. How much performance will it improve if it is calculated only once? Could you make a mirco-benchmark to compare it(The benchmark code does not need to be submitted to pr)?

Reply:
+1 for @LuciferYang 's advice.

Reply:
Thank you for your comments. I make a mirco-benchmark. I used `OpenHashMap` and `HashMap` to calculate the time consumed by the `acquire` and `release` methods of `ResourceAllocator` at the same time. It was found that `OpenHashMap` spent about 5 percent more time than `HashMap`. My test method is to cycle 10000 times, 1000 times, and 100 times respectively.

I'm not sure whether this is reasonable, but the current situation shows that the performance of `OpenHashMap` is not as good as that of `HashMap` in the current case.

Reply:
OK, maybe we shouldn't change this then?
However, if you're looking for optimizations in acquire() and release(), maybe instead of checking contains() and then get()-ing the value, just check once with .getOrElse(address, throw new SparkException(...))

Reply:
I also tested scenarios `create new map`, `apply(key)`, and `contains(key)`, and `OpenHash` didn't show better performance, even slower than `mutable.HashMap`, I think we can delete this TODO If there is no special reason、

 Agree with @srowen, can use `.getOrElse(address, throw new SparkException(...))` save  one `contains`

Reply:
So far, this TODO has not shown good performance after verification. Should we remove it? @LuciferYang @srowen @dongjoon-hyun 

And I tested the `getOrElse` method. Compared with the previous method, it will reduce the time consumption by about% 4. Can I submit code on the current pr ?

Reply:
Yeah I would remove the TODO, and if that tiny optimization helps, then let's just change this JIRA/PR to be about this little improvement.

Reply:
Thank you very much for your suggestion.@srowen It has been changed. :-)